### PR TITLE
feat: support cmake compile

### PR DESCRIPTION
**Pull request by JayInnn**
New Feature:
1. add CMakeLists.txt and related files to support cmake compile;
2. add README.md to explain how to use and note the limitation about this PR.
3. test cmake on Centos and Ubuntu OS.

### CODE REVIEW

**Comment by JayInnn** [comment]
Testing:
image
image

**Comment by wu-sheng** [comment]
@wbpcode Does this fit envoy side requirements?

**Comment by wbpcode** [comment]
Hmmm, additional build system support for the cpp2sky won't effect envoy. So from envoy's point, I think it's OK.

But from the cpp2sky self's point, is these submodule necesssary? If there better way to manager these deps?

**Comment by JayInnn** [comment]
> But from the cpp2sky self's point, is these submodule necesssary? If there better way to manager these deps?

I think I need to explain the reason that this project have added the submodules. 
Reason one: this feat support several ways to compile the project, by using cmake. FetchContent(default way) function will download deps from network, so the submodules has no effects. But add_subdirectory & find_package function need the codes of the deps, which was compiled and installed.

Reason two: when I did this task, due to the grpc archive, I could not figure out the tag of grpc. Therefore I added the submodule and checkouted to the specified branch and noted the version in README.md. As for other deps, perhaps, these should be marked.

**Comment by wbpcode** [comment]
Got it, thanks for you explanation. But basically, IMO, I can only accept the `FetchContent` one because I want to keep the repo clean.

**Comment by JayInnn** [comment]
> Got it, thanks for you explanation. But basically, IMO, I can only accept the `FetchContent` one because I want to keep the repo clean.

**Comment by wbpcode** [comment]
Thanks for you update. Now we will have a clean tree.
Would you mind add the build to the ci job then we can ensure this cmake building basically ok? Thanks.

(And I am so sorry for the delay, I check your new messaage in my phone and forget to respond.

File: README.md
Comment (wu-sheng) [inline README.md:34]:
```suggestion
#### CMake
```

**Comment by JayInnn** [comment]
> Would you mind add the build to the ci job then we can ensure this cmake building basically ok? Thanks.

My pleasure!
I have added cmake-building ci in the workflows. The new feature will work well, i think :)

**Comment by JayInnn** [comment]
image
Some errors ocurred, but the cmake-buiding ci was successful. Maybe, the maintainer should check the error.

**Comment by wu-sheng** [comment]
@wbpcode It seems the coverage fails.

**Comment by wbpcode** [comment]
Weird, it just success yesterday. Let's try it again.

**Review by wbpcode — APPROVED** [review]
Thanks, LGTM. friendly ping @wu-sheng for merge or second review.

**Pull request merged**
